src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -129,15 +134,32 @@ public Task Consume(CancellationToken cancellationToken = default(CancellationTo
                 }
                 else
                 {
+                    var messageBody = lastMessageBody as ForContentLength;
+                    if (messageBody != null)
+                    {
+                        messageBody.Reset(keepAlive, contentLength);
+                    }
+
                     return new ForContentLength(keepAlive, contentLength, context);
                 }
             }
 
             if (keepAlive)
             {
+                var messageBody = lastMessageBody as ForContentLength;
+                if (messageBody != null)
+                {
+                    messageBody.Reset(keepAlive, 0);
```

Comment:
I don't get it... What's the point of resetting this when you're returning a new instance below anyway?